### PR TITLE
Stylo: Add FFI interfaces for exposing style sources.

### PR DESCRIPTION
**Pull request by bradwerth**
https://bugzilla.mozilla.org/show_bug.cgi?id=1346256
https://reviewboard.mozilla.org/r/119044/

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16328)

### CODE REVIEW

**Comment by Manishearth** [comment]
@bors-servo delegate+

use `@bors-servo r=whoever` to approve for landing

**Comment by Manishearth** [comment]
```
error[E0432]: unresolved import `gecko_bindings::bindings::Gecko_GetBaseSize`
   --> C:\buildbot\slave\windows-msvc-dev\build\components\style\gecko\wrapper.rs:460:13
    |
460 |         use gecko_bindings::bindings::Gecko_GetBaseSize;
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `Gecko_GetBaseSize` in `gecko_bindings::bindings`

error[E0432]: unresolved import `gecko_bindings::bindings::Gecko_GetBaseSize`
     --> c:\buildbot\slave\windows-msvc-dev\build\target\geckolib\debug\build\style-d5c8007cf63b1594\out/properties.rs:23716:21
      |
23716 |                 use gecko_bindings::bindings::Gecko_GetBaseSize;
      |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `Gecko_GetBaseSize` in `gecko_bindings::bindings`

error[E0412]: cannot find type `FontSizePrefs` in module `gecko_bindings::structs`
   --> C:\buildbot\slave\windows-msvc-dev\build\components\style\gecko\wrapper.rs:442:45
    |
442 |     pub font_size_cache: RefCell<Vec<(Atom, ::gecko_bindings::structs::FontSizePrefs)>>,
    |                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `gecko_bindings::structs`

error[E0412]: cannot find type `FontSizePrefs` in module `structs`
   --> C:\buildbot\slave\windows-msvc-dev\build\components\style\gecko\wrapper.rs:473:6
    |
473 | impl structs::FontSizePrefs {
    |      ^^^^^^^^^^^^^^^^^^^^^^ not found in `structs`

error[E0531]: cannot find unit struct/variant or constant `kPresContext_DefaultVariableFont_ID` in module `structs`
   --> C:\buildbot\slave\windows-msvc-dev\build\components\style\gecko\wrapper.rs:476:13
    |
476 |             structs::kPresContext_DefaultVariableFont_ID => self.mDefaultVariableSize,
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `structs`

error[E0531]: cannot find unit struct/variant or constant `kPresContext_DefaultFixedFont_ID` in module `structs`
   --> C:\buildbot\slave\windows-msvc-dev\build\components\style\gecko\wrapper.rs:477:13
    |
477 |             structs::kPresContext_DefaultFixedFont_ID => self.mDefaultFixedSize,
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `structs`

error: struct `RawServoStyleRule` is private
  --> C:\buildbot\slave\windows-msvc-dev\build\components\style\gecko\arc_types.rs:12:52
   |
12 | use gecko_bindings::bindings::{RawServoStyleSheet, RawServoStyleRule, RawServoImportRule};
   |                                                    ^^^^^^^^^^^^^^^^^

error: the type of this value must be known in this context
   --> C:\buildbot\slave\windows-msvc-dev\build\components\style\gecko\wrapper.rs:462:30
    |
462 |         if let Some(sizes) = cache.iter().find(|el| el.0 == *font_name) {
    |                              ^^^^^^^^^^^^

error: aborting due to previous error
```

File: components/style/gecko_bindings/bindings.rs [hunk 1]
```diff
@@ -16,18 +16,18 @@ use gecko_bindings::structs::RawGeckoAnimationValueList;
...
+use gecko_bindings::structs::RawGeckoServoStyleRuleList;
 use gecko_bindings::structs::RawGeckoURLExtraData;
 use gecko_bindings::structs::RefPtr;
 use gecko_bindings::structs::CSSPseudoClassType;
-use gecko_bindings::structs::TraversalRestyleBehavior;
 use gecko_bindings::structs::TraversalRootBehavior;
 use gecko_bindings::structs::ComputedTimingFunction_BeforeFlag;
 use gecko_bindings::structs::FontFamilyList;
 use gecko_bindings::structs::FontFamilyType;
-use gecko_bindings::structs::FontSizePrefs;
```
Comment (Manishearth) [inline components/style/gecko_bindings/bindings.rs]: undo this change

**Comment by emilio** [comment]
```
./ports/geckolib/glue.rs:63: use statement is not in alphabetical order
	expected: style::gecko_bindings::structs::RawGeckoPresContextOwned
	found: style::gecko_bindings::structs::{RawServoStyleRule, ServoStyleSheet}
```
